instaloader/instaloader.py

```diff
@@ -1197,14 +1202,13 @@ def download_tagged(self, profile: Profile, fast_update: bool = False,
         posts_to_download: Iterator[Post] = profile.get_tagged_posts()
         if latest_stamps is not None:
             last_scraped = latest_stamps.get_last_tagged_timestamp(profile.username)
-            posts_to_download = takewhile(lambda p: p.date_utc.replace(tzinfo=timezone.utc) > last_scraped,
-                                          posts_to_download)
+            posts_takewhile = lambda p: p.date_utc.replace(tzinfo=timezone.utc) > last_scraped
             scraped_timestamp = datetime.now().astimezone()
         self.posts_download_loop(posts_to_download,
                                  target if target
                                  else (Path(_PostPathFormatter.sanitize_path(profile.username)) /
                                        _PostPathFormatter.sanitize_path(':tagged')),
-                                 fast_update, post_filter)
+                                 fast_update, post_filter, takewhile=posts_takewhile)
```

Comment:
Here is `posts_takewhile` referenced even though it is undefined when `latest_stamps` is None.

(same in the other altered calls to `posts_download_loop`)